flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/HiveScalarFunction.java

```diff
@@ -104,4 +114,66 @@ public Object eval(Object... args) {
 	 * Evaluation logical, args will be wrapped when is a single array.
 	 */
 	protected abstract Object evalInternal(Object[] args);
+
+	private void setArguments(CallContext callContext) {
```

Comment:
Move this to a dedicated class because it will be shared for other functions as well?

Reply:
We can rename `HiveFunction` to `HiveLegacyFunction`, and create a new abstract base class `HiveFunction`.